### ISSUE TITLE
训练从fast-livo2转换过来的包的时候，出现了报错

### ISSUE THREAD

**Issue by llllliu123** [hku-mars/GS-SDF#19]
{ 80.0%}[3998/5000, 1.9it/s, 2074.7<520.0s], loss: 0.113, sstd: 0.00987, s_pt_n: 3.26e+04, s_ray_n: 1.82e+03, s_pts_per_ray: 18, sdf: 0.0753  Conducting draw_loss command: python3 /home/bula/gs_ws/gs_sdf_ws/src/GS-SDF/eval/draw_loss.py -l /home/bula/gs_ws/gs_sdf_ws/src/GS-SDF/output/2025-10-20-16-10-17_fast_livo2_1011.bag_lll.yaml/nsdf_log.txt
 { 80.0%}[4000/5000, 1.9it/s, 2075.8<518.9s], loss: 0.114, sstd: 0.00989, s_pt_n: 3.32e+04, s_ray_n: 1.81e+03, s_pts_per_ray: 18.1, sdf: 0.0756Timing Statistics saved to /home/bula/gs_ws/gs_sdf_ws/src/GS-SDF/output/2025-10-20-16-10-17_fast_livo2_1011.bag_lll.yaml/nsdf_log/nsdf_log.txttiming.txt
 { 80.0%}[4001/5000, 1.9it/s, 2076.3<518.4s], loss: 0.115, sstd: 0.00988, s_pt_n: 3.2e+04, s_ray_n: 1.82e+03, s_pts_per_ray: 18, sdf: 0.0764   
Outlier Removal(0.0821876): 4134901 -> 3593335
 { 99.9%}[4997/5000, 1.9it/s, 2587.1<1.6s], loss: 0.116, sstd: 0.00859, s_pt_n: 3.32e+04, s_ray_n: 1.65e+03, s_pts_per_ray: 19.9, sdf: 0.078   Conducting draw_loss command: python3 /home/bula/gs_ws/gs_sdf_ws/src/GS-SDF/eval/draw_loss.py -l /home/bula/gs_ws/gs_sdf_ws/src/GS-SDF/output/2025-10-20-16-10-17_fast_livo2_1011.bag_lll.yaml/nsdf_log.txt
 {100.0%}[4999/5000, 1.9it/s, 2588.1<0.5s], loss: 0.116, sstd: 0.0084, s_pt_n: 3.31e+04, s_ray_n: 1.65e+03, s_pts_per_ray: 19.8, sdf: 0.0787   Timing Statistics saved to /home/bula/gs_ws/gs_sdf_ws/src/GS-SDF/output/2025-10-20-16-10-17_fast_livo2_1011.bag_lll.yaml/nsdf_log/nsdf_log.txttiming.txt
 {100.0%}[5001/5000, 1.9it/s, 2589.1<0.0s], loss: 0.116, sstd: 0.00853, s_pt_n: 3.27e+04, s_ray_n: 1.65e+03, s_pts_per_ray: 19.9, sdf: 0.0787  [W1020 16:53:40.019189756 TensorShape.cpp:3609] Warning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (function operator())
Marching Cubes {100.0%}[73/72, 93.1it/s, 0.8<0.0s]
Saving mesh to: /home/bula/gs_ws/gs_sdf_ws/src/GS-SDF/output/2025-10-20-16-10-17_fast_livo2_1011.bag_lll.yaml

Export mesh saved to /home/bula/gs_ws/gs_sdf_ws/src/GS-SDF/output/2025-10-20-16-10-17_fast_livo2_1011.bag_lll.yaml/mesh_gs_.ply
[15269888]
TensorOptions(dtype=float, device=cuda:0, layout=Strided, requires_grad=false (default), pinned_memory=false (default), memory_format=(nullopt))
[64, 32]
TensorOptions(dtype=float, device=cuda:0, layout=Strided, requires_grad=false (default), pinned_memory=false (default), memory_format=(nullopt))
[64]
TensorOptions(dtype=float, device=cuda:0, layout=Strided, requires_grad=false (default), pinned_memory=false (default), memory_format=(nullopt))
[64, 64]
TensorOptions(dtype=float, device=cuda:0, layout=Strided, requires_grad=false (default), pinned_memory=false (default), memory_format=(nullopt))
[64]
TensorOptions(dtype=float, device=cuda:0, layout=Strided, requires_grad=false (default), pinned_memory=false (default), memory_format=(nullopt))
[64, 64]
TensorOptions(dtype=float, device=cuda:0, layout=Strided, requires_grad=false (default), pinned_memory=false (default), memory_format=(nullopt))
[64]
TensorOptions(dtype=float, device=cuda:0, layout=Strided, requires_grad=false (default), pinned_memory=false (default), memory_format=(nullopt))
[64, 64]
TensorOptions(dtype=float, device=cuda:0, layout=Strided, requires_grad=false (default), pinned_memory=false (default), memory_format=(nullopt))
[64]
TensorOptions(dtype=float, device=cuda:0, layout=Strided, requires_grad=false (default), pinned_memory=false (default), memory_format=(nullopt))
[2, 64]
TensorOptions(dtype=float, device=cuda:0, layout=Strided, requires_grad=false (default), pinned_memory=false (default), memory_format=(nullopt))
[2]
TensorOptions(dtype=float, device=cuda:0, layout=Strided, requires_grad=false (default), pinned_memory=false (default), memory_format=(nullopt))
[542969, 3]
TensorOptions(dtype=float, device=cuda:0, layout=Strided, requires_grad=false (default), pinned_memory=false (default), memory_format=(nullopt))
[542969, 3]
TensorOptions(dtype=float, device=cuda:0, layout=Strided, requires_grad=false (default), pinned_memory=false (default), memory_format=(nullopt))
[542969, 4]
TensorOptions(dtype=float, device=cuda:0, layout=Strided, requires_grad=false (default), pinned_memory=false (default), memory_format=(nullopt))
[542969]
TensorOptions(dtype=float, device=cuda:0, layout=Strided, requires_grad=false (default), pinned_memory=false (default), memory_format=(nullopt))
[542969, 1, 3]
TensorOptions(dtype=float, device=cuda:0, layout=Strided, requires_grad=false (default), pinned_memory=false (default), memory_format=(nullopt))
[542969, 15, 3]
TensorOptions(dtype=float, device=cuda:0, layout=Strided, requires_grad=false (default), pinned_memory=false (default), memory_format=(nullopt))
[W1020 16:53:42.078607155 anomaly_mode.cpp:49] Warning: Error detected in DivBackward0. Traceback of forward call that caused the error:
frame #0: <unknown function> + 0x4bf23cd (0x7f540b2703cd in /home/bula/gs_ws/libtorch/lib/libtorch_cpu.so)
frame #1: <unknown function> + 0x4c502d8 (0x7f540b2ce2d8 in /home/bula/gs_ws/libtorch/lib/libtorch_cpu.so)
frame #2: <unknown function> + 0x4c50d5b (0x7f540b2ced5b in /home/bula/gs_ws/libtorch/lib/libtorch_cpu.so)
frame #3: at::_ops::div_Tensor::call(at::Tensor const&, at::Tensor const&) + 0x172 (0x7f5408f29e32 in /home/bula/gs_ws/libtorch/lib/libtorch_cpu.so)
frame #4: rasterization_2dgs_sdf(at::Tensor const&, at::Tensor const&, at::Tensor const&, at::Tensor const&, at::Tensor const&, at::Tensor const&, at::Tensor const&, int, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, float, float, float, std::optional<int>, bool, int, std::optional<at::Tensor>, bool, bool, bool, std::shared_ptr<LocalMap> const&, bool const&) + 0xc2b (0x7f53a43668ab in /home/bula/gs_ws/gs_sdf_ws/devel/lib/libneural_gaussian.so)
frame #5: NeuralGS::render[abi:cxx11](at::Tensor const&, sensor::Cameras, bool const&, int const&) + 0xc5b (0x7f53a4368a4b in /home/bula/gs_ws/gs_sdf_ws/devel/lib/libneural_gaussian.so)
frame #6: NeuralSLAM::render_image[abi:cxx11](int const&, int const&) + 0x414 (0x7f541f71b7e4 in /home/bula/gs_ws/gs_sdf_ws/devel/lib/libneural_mapping_lib.so)
frame #7: NeuralSLAM::gs_train_batch_iter[abi:cxx11](int const&, bool const&) + 0x78b (0x7f541f725a8b in /home/bula/gs_ws/gs_sdf_ws/devel/lib/libneural_mapping_lib.so)
frame #8: NeuralSLAM::gs_train(int) + 0x2053 (0x7f541f728e33 in /home/bula/gs_ws/gs_sdf_ws/devel/lib/libneural_mapping_lib.so)
frame #9: NeuralSLAM::batch_train() + 0xa57 (0x7f541f72b707 in /home/bula/gs_ws/gs_sdf_ws/devel/lib/libneural_mapping_lib.so)
frame #10: <unknown function> + 0xd6df4 (0x7f53a4702df4 in /lib/x86_64-linux-gnu/libstdc++.so.6)
frame #11: <unknown function> + 0x8609 (0x7f541f4cd609 in /lib/x86_64-linux-gnu/libpthread.so.0)
frame #12: clone + 0x43 (0x7f53a453c353 in /lib/x86_64-linux-gnu/libc.so.6)
 (function print_stack)
terminate called after throwing an instance of 'std::runtime_error'
  what():  Function 'DivBackward0' returned nan values in its 0th output.
Stack trace (most recent call last) in thread 55921:
#16   Object "", at 0xffffffffffffffff, in 
#15   Source "../sysdeps/unix/sysv/linux/x86_64/clone.S", line 95, in __clone [0x7f53a453c352]
#14   Source "/build/glibc-B3wQXB/glibc-2.31/nptl/pthread_create.c", line 477, in start_thread [0x7f541f4cd608]
#13   Object "/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.28", at 0x7f53a4702df3, in 
#12   Source "/home/bula/gs_ws/gs_sdf_ws/src/GS-SDF/include/neural_mapping/neural_mapping.cpp", line 849, in batch_train [0x7f541f72b706]
        846:         std::cout << param.options() << "\n";
        847:       }
        848:     }
      > 849:     gs_train(k_gs_iter_step);
        850:   }
        851: 
        852:   p_train->toc_sum();
#11 | Source "/home/bula/gs_ws/gs_sdf_ws/src/GS-SDF/include/neural_mapping/neural_mapping.cpp", line 377, in backward
    |   375:       auto [gs_loss, render_results] = gs_train_batch_iter(i, false);
    |   376:       p_optimizer_->zero_grad();
    | > 377:       gs_loss.backward();
    |   378:       p_optimizer_->step();
    |   379:     }
      Source "/home/bula/gs_ws/libtorch/include/ATen/core/TensorBody.h", line 445, in gs_train [0x7f541f728e84]
        442:       TORCH_CHECK(inputs.value().size() > 0, "'inputs' argument to backward cannot be empty")
        443:       this->_backward(inputs.value(), gradient, retain_graph, create_graph);
        444:     } else {
      > 445:       this->_backward({}, gradient, retain_graph, create_graph);
        446:     }
        447:   }
#10   Object "/home/bula/gs_ws/libtorch/lib/libtorch_cpu.so", at 0x7f54082479ab, in at::Tensor::_backward(c10::ArrayRef<at::Tensor>, std::optional<at::Tensor> const&, std::optional<bool>, bool) const
#9    Object "/home/bula/gs_ws/libtorch/lib/libtorch_cpu.so", at 0x7f540bf0679c, in torch::autograd::VariableHooks::_backward(at::Tensor const&, c10::ArrayRef<at::Tensor>, std::optional<at::Tensor> const&, std::optional<bool>, bool) const
#8    Object "/home/bula/gs_ws/libtorch/lib/libtorch_cpu.so", at 0x7f540be9a8d9, in torch::autograd::backward(std::vector<at::Tensor, std::allocator<at::Tensor> > const&, std::vector<at::Tensor, std::allocator<at::Tensor> > const&, std::optional<bool>, bool, std::vector<at::Tensor, std::allocator<at::Tensor> > const&)
#7    Object "/home/bula/gs_ws/libtorch/lib/libtorch_cpu.so", at 0x7f540be97b3c, in torch::autograd::run_backward(std::vector<at::Tensor, std::allocator<at::Tensor> > const&, std::vector<at::Tensor, std::allocator<at::Tensor> > const&, bool, bool, std::vector<at::Tensor, std::allocator<at::Tensor> > const&, bool, bool)
#6    Object "/home/bula/gs_ws/libtorch/lib/libtorch_cpu.so", at 0x7f540beb7e6e, in torch::autograd::Engine::execute(std::vector<torch::autograd::Edge, std::allocator<torch::autograd::Edge> > const&, std::vector<at::Tensor, std::allocator<at::Tensor> > const&, bool, bool, bool, std::vector<torch::autograd::Edge, std::allocator<torch::autograd::Edge> > const&)
#5    Object "/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.28", at 0x7f53a46d636e, in std::rethrow_exception(std::__exception_ptr::exception_ptr)
#4    Object "/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.28", at 0x7f53a46d63e6, in std::terminate()
#3    Object "/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.28", at 0x7f53a46d637b, in 
#2    Object "/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.28", at 0x7f53a46ca8d0, in 
#1    Source "/build/glibc-B3wQXB/glibc-2.31/stdlib/abort.c", line 79, in abort [0x7f53a443f858]
#0    Source "../sysdeps/unix/sysv/linux/raise.c", line 51, in raise [0x7f53a446000b]
Aborted (Signal sent by tkill() 55795 1000)
Aborted (core dumped)
我按照指示录制了修改版fast-livo2的bag，然后在运行的时候出现了上面的错误。能帮我看一看是什么问题吗？

**Comment by llllliu123** [comment]
解决问题了。主要修改的地方是两个，一个是因为我使用的是mid360雷达，它的点云会比avia雷达更加的稀疏。最好自己去ply里面看看，调整一下降采样点云的个数。然后对于个人笔记本显存不够的问题，可以调整参数进行解决。作者大大都有提供参数注释。优秀的作品。
